src/main/java/com/welovecoding/nbeditorconfig/io/writer/StyledDocumentWriter.java

```diff
@@ -157,16 +160,28 @@ public void run() {
 
                 // Reformat code (to apply ident size & styles)
                 // TODO: Do this only if CodeStylePreferences have been changed
-                Reformat reformat = Reformat.get(cookie.getDocument());
-                reformat.lock();
-                try {
-                  reformat.reformat(0, cookie.getDocument().getLength());
-                } catch (BadLocationException ex) {
-                  LOG.log(Level.SEVERE, "AutoFormat on document not possible: {0}", ex.getMessage());
-                } finally {
-                  reformat.unlock();
-                  // Save document after reformat
-                  cookie.saveDocument();
+
+                Preferences prefs = MimeLookup.getLookup(DocumentUtilities.getMimeType(cookie.getDocument())).lookup(Preferences.class);
+
+                if (prefs.getBoolean(SimpleValueNames.ON_SAVE_USE_GLOBAL_SETTINGS, true)) {
+                    prefs = MimeLookup.getLookup(MimePath.EMPTY).lookup(Preferences.class);
+                }
+
+                String policy = prefs.get(SimpleValueNames.ON_SAVE_REFORMAT, null);
+
+                if (!"never".equals(policy)) {
```

Comment:
Is the string "`never`" available as a constant from the NetBeans API? I would also prefer having `policy.equals("never")` than the other way around.

Reply:
The string "never" is set in this file: `C:\Users\Chrl\AppData\Roaming\NetBeans\8.2\config\Editors\Preferences\org-netbeans-modules-editor-settings-CustomPreferences.xml`

```XML
    <entry javaType="java.lang.String" name="on-save-reformat" xml:space="preserve">
        <value><![CDATA[never]]></value>
    </entry>
```

It comes from NetBeans. I don't know where you can find it in an other way.


I copied the code from here: http://grepcode.com/file/bits.netbeans.org/maven2/org.netbeans.api/org-netbeans-modules-editor/RELEASE80/org/netbeans/modules/editor/impl/ReformatBeforeSaveTask.java#114

but sure, I can change it, if you want it. Let me know it.

Reply:
Ok, since this is core NetBeans code it should be fine this way. 👍  Thank you so much for your contribution!

Reply:
Great. Thx :)